src/components/TodoApp.tsx

```diff
@@ -0,0 +1,130 @@
+import { useState, useContext } from 'react';
+import { DispatchContex, TodosContext } from './TodosContext';
+import { TodoList } from './TodoList';
+import { TodosFilter } from './TodosFilter';
+import { ToDo } from '../types/todo';
+
+export const TodoApp: React.FC = () => {
+  const dispatch = useContext(DispatchContex);
+  const { todos, filtred } = useContext(TodosContext);
+
+  const [value, setValue] = useState('');
+  const trimedValue = value.trim();
+
+  const addNewToDo = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (trimedValue === '') {
+      return;
+    }
+
+    const newTodo = {
+      id: +new Date(),
+      title: trimedValue,
+      completed: false,
+    };
+
+    if (todos.length > 0) {
```

Comment:
```suggestion
    if (!!todos.length) {
```
**PLS DO THE SAME FOR ALL SUCH CASES IN THE PROJECT**